### ISSUE TITLE
macOS: Building profiler runtime with system clang 15 gives `warning: argument unused during compilation: '-stdlib=libc++'`

### ISSUE THREAD

**Issue by Zalathar** [rust-lang/rust#116552]
On macOS, if I have `profiler = true` in my `config.toml` and I run `x build library`, the build succeeds but I see dozens of console warnings:

```
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
```

<details>
<summary>(full console output)</summary>

```
$ x build library
Building bootstrap
    Finished dev [unoptimized] target(s) in 0.03s
Building stage0 library artifacts (aarch64-apple-darwin)
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
    Finished release [optimized] target(s) in 0.07s
Building compiler artifacts (stage0 -> stage1, aarch64-apple-darwin)
    Finished release [optimized] target(s) in 0.13s
Creating a sysroot for stage1 compiler (use `rustup toolchain link 'name' build/host/stage1`)
Building stage1 library artifacts (aarch64-apple-darwin)
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
warning: clang: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
    Finished release [optimized] target(s) in 0.07s
Build completed successfully in 0:00:01
```
</details>

This started happening after I accepted a system update to *Command Line Tools for Xcode* version 15.0.

The warnings appear to occur when building `library/profiler_runtime`. This component is not built by default (hence the `profiler = true`), but is needed when building coverage-instrumented binaries, and is included in actual releases of `rustc`.

---

### System info

```
$ sw_vers
ProductName:		macOS
ProductVersion:		13.6
BuildVersion:		22G120

$ clang --version
Apple clang version 15.0.0 (clang-1500.0.40.1)
Target: arm64-apple-darwin22.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

```toml
# config.toml
profile = "compiler"
change-id = 115898

[build]
profiler = true
```

**Comment by Zalathar** [comment]
I tracked this down to a C++ flag being unconditionally added to `CFLAGS`, and wrote a candidate fix in #116017, but I'm not sure what needs to be done to move forward with that.

**Comment by Zalathar** [comment]
The same warning also appears when building C files as part of building LLVM from source.

**Comment by Zalathar** [comment]
This was fixed by #116017.
